exasol_transformers_extension/deploy.py

```diff
@@ -1,29 +1,86 @@
-    import language_container_deployer_main, slc_parameter_formatters, CustomizableParameters
-from exasol_transformers_extension.deployment.te_language_container_deployer import TeLanguageContainerDeployer
+from exasol.python_extension_common.cli.std_options import (
+    StdParams, StdTags, select_std_options, ParameterFormatters)
+from exasol.python_extension_common.cli.language_container_deployer_cli import (
+    LanguageContainerDeployerCli)
+from exasol.python_extension_common.cli.bucketfs_conn_object_cli import BucketfsConnObjectCli
+from exasol.python_extension_common.connections.pyexasol_connection import open_pyexasol_connection
+from exasol.python_extension_common.connections.bucketfs_location import (
+    ConnectionInfo, write_bucketfs_conn_object)
+from exasol_transformers_extension.deployment.te_language_container_deployer import (
+    TeLanguageContainerDeployer)
+from exasol_transformers_extension.deployment.scripts_deployer import ScriptsDeployer
 
+DEPLOY_SLC_ARG = 'deploy_slc'
+DEPLOY_SCRIPTS_ARG = 'deploy_scripts'
+CONTAINER_URL_ARG = 'container_url'
+CONTAINER_NAME_ARG = 'container_name'
+BUCKETFS_CONN_NAME_ARG = 'bucketfs_conn_name'
+TOKEN_CONN_NAME_ARG = 'token_conn_name'
+TOKEN_ARG = 'token'
 
-@click.group()
-def main():
-    pass
+ver_formatter = ParameterFormatters()
+ver_formatter.set_formatter(CONTAINER_URL_ARG, TeLanguageContainerDeployer.SLC_URL_FORMATTER)
+ver_formatter.set_formatter(CONTAINER_NAME_ARG, TeLanguageContainerDeployer.SLC_NAME)
+formatters = {StdParams.version: ver_formatter}
 
 
-slc_parameter_formatters.set_formatter(CustomizableParameters.container_url,
-                                       TeLanguageContainerDeployer.SLC_URL_FORMATTER)
-slc_parameter_formatters.set_formatter(CustomizableParameters.container_name,
-                                       TeLanguageContainerDeployer.SLC_NAME)
+def get_opt_name(arg_name: str) -> str:
+    # This and the next function should have been implemented in the PEC.
+    return f'--{arg_name.replace("_", "-")}'
 
-main.add_command(scripts_deployer_main)
-main.add_command(language_container_deployer_main)
+
+def get_bool_opt_name(arg_name: str) -> str:
+    opt_name = arg_name.replace("_", "-")
+    return f'--{opt_name}/--no-{opt_name}'
+
+
+opts = select_std_options([StdTags.DB, StdTags.BFS, StdTags.SLC], formatters=formatters)
+opts.append(click.Option([get_bool_opt_name(DEPLOY_SLC_ARG)], type=bool, default=True))
+opts.append(click.Option([get_bool_opt_name(DEPLOY_SCRIPTS_ARG)], type=bool, default=True))
+opts.append(click.Option([get_opt_name(BUCKETFS_CONN_NAME_ARG)], type=str))
+opts.append(click.Option([get_opt_name(TOKEN_CONN_NAME_ARG)], type=str))
+opts.append(click.Option([get_opt_name(TOKEN_ARG)], type=str))
+
+
+def deploy(**kwargs):
+
+    # Make sure there is a valid language_alias
+    if not kwargs.get(StdParams.language_alias.name):
+        kwargs[StdParams.language_alias.name] = 'PYTHON3_TE'
```

Comment:
This may indicate an optional future improvement, i.e. specifying application-default values for specific CLI commands. Or maybe variable `opts` above allows to modify default values, too?

Reply:
Good point. I can actually override the definition of a standard option. The deficiency in PEC I can see now is that it simply replaces one definition with another. It could be smarter and merge the two definitions. For example, here I just want to override the default value. I don't want to change the help string etc.

Reply:
:+1: